### PR TITLE
feat(ai): fix visualization planning, JSON decoding, and regional country group resolution

### PR DESCRIPTION
**Pull request by rafmacalaba**
## Description
This PR addresses several critical agent and prompt issues observed during testing:

1. **Visualization Planning False GAPs**:
   - Added context-level rules to the Research Agent system prompt explaining that visualization tools are supported and handled downstream by the Narrator node.
   - Prevented the Research Agent from generating false GAPs on map/chart requests.
   - Added `chart_type` parameter to the `### VIZ_PLAN:` section of the prompt.
   - Instructed the Narrator Agent system prompt to explicitly pass `chart_type="map"` or `chart_type="choropleth"` to the visualization spec tool on spatial queries.

2. **JSON Normalization Graceful Fallback**:
   - Wrapped `json.loads` in `tool_output_normalize.py` with a robust try-except handler to safely fallback to returning plain text instead of crashing the SSE stream when tools return plain text.
   - Added regression unit tests in `tests/test_tool_output_normalize.py` covering all edge cases.

3. **Regional Country Group Hardcoding Resolution**:
   - Updated the Narrator node tool whitelist in `narrator.py` to allow passing the raw results of lookup/expansion tools (`data360_expand_country_group`, `data360_get_disaggregation`, `data360_find_codelist_value`, `data360_search_indicators`) to the Writer's prompt.
   - Added a defensible architectural reason as code comments justifying their inclusion in the `data_tools` whitelist (as these tools serve as 'data-bearing' tools for metadata queries).
   - Removed hardcoded database regional listings (such as South Asia) from the system prompts and local `geo_utils.py` dictionary to defer strictly to the dynamic `data360_expand_country_group` tool.

4. **Two-Country Query Routing Optimization**:
   - Added strict prompt guidelines forcing the agent to use `data360_compare_countries` (which renders a side-by-side comparison card) instead of `data360_get_data` (which is limited to a single fact card) for exactly 2 countries.
   - Restricts `data360_compare_countries` to exactly 2 countries due to UI card rendering limitations, preferring batched `data360_get_data` or `data360_summarize_data` for 3+ countries.
image

### CODE REVIEW

File: backend/app/ai/prompts.py [hunk 1]
```diff
@@ -264,16 +264,13 @@ def get_research_agent_system_prompt() -> str:
...
-- ASEAN: PHL, IDN, VNM, THA, MYS, MMR, KHM, LAO, SGP, BRN
-- South Asia: BGD, IND, PAK, NPL, LKA, AFG, MDV, BTN
-- Sub-Saharan Africa: NGA, ETH, KEN, GHA, TZA, UGA, ZAF, MOZ, SEN, ZMB
-- MENA: EGY, MAR, TUN, DZA, JOR, LBN, IRQ, YEM, SAU, ARE
-- Latin America: BRA, MEX, COL, ARG, PER, CHL, ECU, BOL
-- East Asia: CHN, IDN, PHL, VNM, THA, MYS, KHM, MMR
-- Europe & Central Asia: TUR, KAZ, UKR, UZB, GEO, ARM, MDA, ALB
+- Database Region/Income Groups: If the user mentions a regional group (e.g. South Asia, Sub-Saharan Africa, East Asia, MENA, Latin America, Europe & Central Asia) or income group, ALWAYS call data360_expand_country_group (with group_code like SAS, SSF, EAS, MEA, LAC, ECS, LIC, etc.) to fetch the exact database membership dynamically.
+- Non-Database Groups: For groups that cannot be expanded via a single database group code, resolve using custom mappings:
+  - ASEAN: PHL, IDN, VNM, THA, MYS, MMR, KHM, LAO, SGP, BRN
```
Comment (avsolatorio) [inline backend/app/ai/prompts.py:274]: This is not correct. ASEAN now includes Timor-Leste.

Also, if we want to expose the list of codes here, it would be good to have it reference the ones in `geo_utils.py`, so you only need to maintain the list there. Think of having:

```python
f"""
- ASEAN: {', '.join(geo_utils['asean'])}
"""
```

I was also thinking of using `pycountry` or `countryguess`, but they're both copy-left.

**Review by avsolatorio — COMMENTED** [review]
Hello @rafmacalaba , please check one comment below.

File: backend/app/ai/prompts.py (hunk 1 above)
Comment (rafmacalaba) [reply backend/app/ai/prompts.py:274]: missed your comment XD will implement this @avsolatorio, I have implemented this by dynamically referencing the codes from geo_utils.py inside the prompts, and added Timor-Leste (TLS) to the ASEAN list. I will not use pycountry and countryguess because of the license.
